iati/core/tests/test_resources.py

```diff
@@ -2,14 +2,35 @@
 from lxml import etree
 import pytest
 import six
+import iati.core.constants
 import iati.core.resources
 
 
 class TestResources(object):
     """A container for tests relating to resources."""
 
+    @pytest.fixture(params=[None] + iati.core.constants.STANDARD_VERSIONS)
+    def standard_version_optional(self, request):
+        """Return a list that can be used as variable positional parameters (i.e. `*standard_version_optional`) to test.
```

Comment:
What is the benefit of returning a list rather than `None` or a string? It seems to confuse things since the result then needs unpacking.

Reply:
In the case of `test_get_folder_path_for_version`, we want to test that `iati.core.resources.get_folder_path_for_version` works as expected in the following circumstances:

```
iati.core.resources.get_folder_path_for_version('2.02')
iati.core.resources.get_folder_path_for_version('2.01')
iati.core.resources.get_folder_path_for_version('1.05')
iati.core.resources.get_folder_path_for_version('1.04')
iati.core.resources.get_folder_path_for_version()
iati.core.resources.get_folder_path_for_version(None)
```

At present, this fixture is designed to provide data to test all but the last - i.e. the first 4 cases are covered by paramererisation of `iati.core.constants.STANDARD_VERSIONS`, then returning an empty list means no args will be passes.

The last case isn't covered at all yet, but I plan to change the logic in to enable this fixture to return `[None]` in a future commit.

Reply:
Resolved in a4722b0